### PR TITLE
Removed needsBus and added/fixed tests for travel

### PR DESCRIPTION
**Pull request by RohitGarudadri**
### Tickets:

HCK-194, 176

### List of changes:

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

- Removed all instances of 'needsBus' from backend, as well as added tests for the travel field.

## Type of change

Please delete options that are not relevant.

-   [x] Bug fix (non-breaking change which fixes an issue)

### How has this been tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

-   [x] Added a new test in the hacker.test.js

### Checklist:

-   [x] My code follows the style guidelines of this project
-   [x] I have performed a self-review of my own code
-   [x] I have commented my code, particularly in hard-to-understand areas
-   [x] My changes generate no new warnings
-   [x] Listed change(s) in the Changelog
-   [x] I have added tests that prove my fix is effective or that my feature works
-   [x] New and existing unit tests pass locally with my changes
-   [x] I have made corresponding changes to the documentation
-   [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: middlewares/validators/validator.helper.js [hunk 1]
```diff
@@ -41,7 +41,7 @@ function integerValidator(
                 return value >= lowerBound && value <= upperBound;
             })
             .withMessage(
-                `${fieldname} must be between ${lowerBound} and  ${upperBound}`
+                `${fieldname} must be between ${lowerBound} and ${upperBound}`
```
Comment (loreina) [inline middlewares/validators/validator.helper.js]: does this only validate that travel is an integer? or does it also cover the [0,100] inclusive bound?

File: tests/util/hacker.test.util.js [hunk 2]
```diff
@@ -391,7 +391,46 @@ const invalidHacker1 = {
...
             dietaryRestrictions: ["Gluten-Free"],
             shirtSize: "L",
-            travel: 0
+            travel: "sfg"
+        }
+    }
+};
+
+// duplicate of newHack0, but with 101 for travel
+const invalidHacker2 = {
```
Comment (loreina) [inline tests/util/hacker.test.util.js]: are we missing tests for `impairments` and `barriers`? I'll make a ticket if we are

Comment (RohitGarudadri) [reply tests/util/hacker.test.util.js]: Yes, there are no tests for those currently.

File: middlewares/validators/validator.helper.js (hunk 1 above)
Comment (RohitGarudadri) [reply middlewares/validators/validator.helper.js]: I make a new test all together that also validates the 0-100 bound

Comment (Ty-Won) [reply middlewares/validators/validator.helper.js]: does it validate float vs int? or is that being done on front end?

Comment (loreina) [reply middlewares/validators/validator.helper.js]: @Ty-Won that would be a frontend validation, I'll have a PR for that tn 😉

Comment (krubenok) [reply middlewares/validators/validator.helper.js]: If we want to stay properly RESTful the int validation should be on the API

Comment (pierreTklein) [reply middlewares/validators/validator.helper.js]: Agreed we should have api validation

Comment (RohitGarudadri) [reply middlewares/validators/validator.helper.js]: The validator does check .isInt() which seems to work. If I input a float, like 10.5 for travel, then indeed the error appears saying that it should be an integer. Do we need more than this?

Comment (loreina) [reply middlewares/validators/validator.helper.js]: Seems good to me, I also added int validation on the frontend: https://github.com/hackmcgill/dashboard/pull/764

**Pull request merged**